services/lgtm/lgtm-alerts.service.js

```diff
@@ -6,8 +6,8 @@ const LgtmBaseService = require('./lgtm-base')
 module.exports = class LgtmAlerts extends LgtmBaseService {
   static get route() {
     return {
-      base: 'lgtm/alerts/g',
-      pattern: ':user/:repo',
+      base: 'lgtm/alerts',
+      pattern: ':host/:user/:repo',
```

Comment:
Do you know the set of allowed hosts that LGTM supports? I'm guessing GitHub, Bitbucket, and GitLab which are expressed via `g`, `b`, and probably `gl`, respectively. 

It would be better to scope the `:host` param accordingly (i.e. `:host(g|b|bl)/...`) so that it will display properly in the UI window. Personally, I'd prefer that we actually use `:host(github|bitbucket|gitlab)/...` in the route to present the full name to end users, and internally map that to the corresponding letter/key we need for the API

Reply:
Yes, I believe `g`, `b` and `gl` are the supported VCS hosts. Adding an internal mapping and a fixed list of hosts to display in the UI sounds good. I think for the actual routing it would make sense to fall back to just using whatever the input was to look up the project if it is not in the map. This would have the benefit of keeping existing badges working and it would allow for people to generate URLs manually if more hosts are added in the future, even if the map is not kept up to date.

It doesn't look like this would be too difficult to do, I'll try and update this pull request later today with if there are no objections to this implementation.

Reply:
Sounds like a plan! Regarding some of the specifics on implementation:

> I think for the actual routing it would make sense to fall back to just using whatever the input was to look up the project if it is not in the map

Currently the only existing route that is in use is `/lgtm/(alerts or grade)/g/...`, so if we update the routes to `/lgtm/alerts/:host(github|bitbucket|gitlab/...` then all we need to maintain backwards compatibility is to add a redirector to map `g` to `github`. 

We've got a special service class type that handles this redirect functionality that can be created via the `redirector` helper function. Here's an idea of what that might look like here for LGTM:

```js
'use strict'

const { redirector } = require('..')

const commonAttrs = {
  category: 'analysis',
  dateAdded: new Date('2019-04-26'),
}

module.exports = [
  redirector({
    route: {
      base: 'lgtm/alerts/g',
      pattern: ':user/:repo',
    },
    transformPath: ({ user, repo }) => `/lgtm/alerts/github/${user}/${repo}`,
    ...commonAttrs,
  }),
  redirector({
    route: {
      base: 'lgtm/grade/g',
      pattern: ':user/:repo',
    },
    transformPath: ({ user, repo }) => `/lgtm/grade/github/${user}/${repo}`,
    ...commonAttrs,
  }),
]
```